### DISCUSSION TITLE
Resolved: Flatten `util` and consider what needs to go into the next release

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1418] [General]
Ok crew, we have almost flattened `util` and soon it can be removed. PRs remaining are:

- [ ] https://github.com/rust-bitcoin/rust-bitcoin/pull/1417
- [x] https://github.com/rust-bitcoin/rust-bitcoin/pull/1373
- [x] https://github.com/rust-bitcoin/rust-bitcoin/pull/1260

Flattening `util` was the current task in order for us to get to a place where we can discuss the crate smashing plan.

If we are aiming for a release six months after the last release then we have two months to go. Since crate smashing will take an eon we should probably start thinking about were we draw the line for the next release.

**Comment by tcharding** [comment]
Since we have a bit of time till next release but no where near enough to discuss and do crate smashing should we do the MSRV bump to 1.48 now?

- https://github.com/rust-bitcoin/rust-bitcoin/discussions/1329
- https://github.com/rust-bitcoin/rust-bitcoin/issues/1427

**Reply by apoelstra** [reply]
In the 1.48 discussion issue @TheBlueMatt pointed out that if we wait a few more months, debian will have a new release, shifting 1.41 further into history (out of oldstable). It is not actually one of our MSRV requirements to support debian oldstable, but it would be nice.

**Reply by Kixunil** [reply]
I propose that the upcoming release is 1.41.1 and we bump the *repository* to 1.48 immediately after release. I'm pretty sure it'll take us a while to release the next one so at the time of release there's a high chance Debian Bookworm will be out.

**Reply by apoelstra** [reply]
ACK from me. cc @TheBlueMatt ?

**Comment by apoelstra** [comment]
Regarding crate smashing, sorry, I'll try to reply to this soon.

**Comment by tcharding** [comment]
Not sure how to mark discussions as 'resolved', or if this is even necessary?

**Reply by Kixunil** [reply]
I'm not sure if necessary but I just edited the title to say resolved.

**Reply by tcharding** [reply]
Good thinking.
